framework/src/operators/groupby_batch.rs

```diff
@@ -67,13 +67,21 @@ where
             match item {
                 Ok(packet) => {
                     let key = (self.selector)(&packet);
-                    match self.groups.get_mut(&key) {
+                    match self.groups.get_mut(&Some(key)) {
                         Some(group) => {
                             self.producer.enqueue(packet);
                             group.next().unwrap()
                         }
-                        // can't find the group, drop the packet
-                        None => Err(PacketError::Drop(packet.mbuf())),
+                        None => {
+                            // handle `_` pass-through case
+                            if let Some(group) = self.groups.get_mut(&None) {
```

Comment:
also would be nice to avoid this extra lookup.

Reply:
@drunkirishcoder separate pattern match, you mean? If so, yes. Still, we’d needs a contains_key check if None is in the map though.